scrapy/contrib/loader/processor.py

```diff
@@ -63,6 +63,33 @@ def __call__(self, values):
         return values
 
 
+class JmesProcessor(object):
+    """
+        Query the input string for the jmespath (given at instantiation), and return the answer
+        Requires : jmespath(https://github.com/boto/jmespath), ast
+        Note: JmesProcessor accepts only a single string at a time and returns string/dict/None based on the jmespath query.
+    """
+    def __init__(self, json_path):
+        self.json_path = json_path
+        try:
+            import jmespath
+            self.compiled_path = jmespath.compile(self.json_path)
+        except:
+            raise ImportError("You need to have jmespath(https://github.com/boto/jmespath)"
+                              + " and AST modules before being able to use this processor")
```

Comment:
This is bad for several reasons:

1. If "import jmespath" fails, it hides the original exception which can provide more information about why isn't jmespath available. See e.g. https://github.com/scrapy/scrapy/pull/902.
2. If jmespath.compile fails the code raises an obscure ImportError, hiding the original exception again.
3. `except:` shouldn't be used; it catches e.g. KeyboardInterrupt and SystemExit exceptions.

I suggest to remove try-except entirely. It will fail with "ImportError" pointing to jmespath, this is self-explanatory.